editor/src/components/canvas/canvas-strategies/relative-move-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,457 @@
+import { fireEvent } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { offsetPoint, windowPoint, WindowPoint } from '../../../core/shared/math-utils'
+import { CanvasControlsContainerID } from '../controls/new-canvas-controls'
+import {
+  EditorRenderResult,
+  getPrintedUiJsCode,
+  makeTestProjectCodeWithSnippet,
+  renderTestEditorWithCode,
+} from '../ui-jsx.test-utils'
+
+function dragElement(
+  canvasControl: HTMLElement,
+  startPoint: WindowPoint,
+  dragDelta: WindowPoint,
+  shiftPressed: boolean,
+) {
+  const endPoint = offsetPoint(startPoint, dragDelta)
+  fireEvent(
+    canvasControl,
+    new MouseEvent('mousedown', {
+      bubbles: true,
+      cancelable: true,
+      metaKey: true,
+      shiftKey: shiftPressed,
+      clientX: startPoint.x,
+      clientY: startPoint.y,
+      buttons: 1,
+    }),
+  )
+
+  fireEvent(
+    canvasControl,
+    new MouseEvent('mousemove', {
+      bubbles: true,
+      cancelable: true,
+      shiftKey: shiftPressed,
+      clientX: endPoint.x,
+      clientY: endPoint.y,
+      buttons: 1,
+    }),
+  )
+
+  fireEvent(
+    window,
+    new MouseEvent('mouseup', {
+      bubbles: true,
+      cancelable: true,
+      shiftKey: shiftPressed,
+      clientX: endPoint.x,
+      clientY: endPoint.y,
+    }),
+  )
+}
+
+const setupAndDrag = async (
+  project: string,
+  elementId: string,
+  x: number,
+  y: number,
+  modifiers?: { shiftPressed?: boolean },
+): Promise<EditorRenderResult> => {
+  const renderResult = await renderTestEditorWithCode(project, 'await-first-dom-report')
+  return drag(renderResult, elementId, x, y, modifiers)
+}
+
+const drag = async (
+  renderResult: EditorRenderResult,
+  elementId: string,
+  x: number,
+  y: number,
+  modifiers?: { shiftPressed?: boolean },
+): Promise<EditorRenderResult> => {
+  const targetElement = renderResult.renderedDOM.getByTestId(elementId)
+  const targetElementBounds = targetElement.getBoundingClientRect()
+  const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+  const startPoint = windowPoint({ x: targetElementBounds.x + 5, y: targetElementBounds.y + 5 })
+  const dragDelta = windowPoint({ x, y })
+
+  act(() => dragElement(canvasControlsLayer, startPoint, dragDelta, !!modifiers?.shiftPressed))
```

Comment:
I think here you can use the helper `mouseDragFromPointWithDelta` or `mouseDragFromPointToPoint`

Reply:
oh nice, I did not spot those

Reply:
https://github.com/concrete-utopia/utopia/pull/2604/commits/c71735577000ef8b3bb16d46cac415eadafec617